modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionListener.java

```diff
@@ -88,25 +89,31 @@ public class PartitionListener implements RaftGroupListener {
     /** Storage index tracker. */
     private final PendingComparableValuesTracker<Long> storageIndexTracker;
 
+    /** Low watermark supplier, will return {@code null} if no low watermark has been assigned yet. */
+    private final Supplier<HybridTimestamp> lowWatermarkSupplier;
+
     /**
      * The constructor.
      *
      * @param partitionDataStorage The storage.
      * @param safeTime Safe time tracker.
      * @param storageIndexTracker Storage index tracker.
+     * @param lowWatermarkSupplier Low watermark supplier, will return {@code null} if no low watermark has been assigned yet.
      */
     public PartitionListener(
             PartitionDataStorage partitionDataStorage,
             StorageUpdateHandler storageUpdateHandler,
             TxStateStorage txStateStorage,
             PendingComparableValuesTracker<HybridTimestamp> safeTime,
-            PendingComparableValuesTracker<Long> storageIndexTracker
+            PendingComparableValuesTracker<Long> storageIndexTracker,
+            Supplier<HybridTimestamp> lowWatermarkSupplier
```

Comment:
Can we stop using suppliers? It's annoying. When seeing  `() -> null` in constructor calls, I don't understand what it is.
Why can't we have a dedicated component/interface, that would provide the access to LW?

Reply:
Try to fix it.